metatron/src/main/java/uk/gov/ida/eidas/metatron/core/dto/EidasConfig.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.ida.eidas.metatron.core.dto;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.security.KeyStore;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+
+public class EidasConfig {
+    @JsonProperty
+    private List<EidasCountryConfig> countries = new ArrayList<>();
+
+    @JsonProperty
+    private KeyStore truststore;
+
+    public Collection<EidasCountryConfig> getCountries() {
+        return countries;
+    }
+
+    public KeyStore getKeyStore() {
+        return this.truststore;
+    }
+
+    public void retainAll(Function<EidasConfig, Collection<EidasCountryConfig>> filterFunction) {
```

Comment:
I'm not sure if this is duplicated from other patterns elsewhere, but if not perhaps it could be `retainAllWhere` or `retainWhere` or `dropUnless` (for clarity)? `retainAll` suggests it's keeping everything no matter what. (Or it does to me.)

Reply:
It's what's used by the Collections stuff IIRC.

Reply:
i think it comes from this: https://www.geeksforgeeks.org/arraylist-retainall-method-in-java/